lemminx-maven/src/main/java/org/eclipse/lemminx/maven/searcher/RemoteRepositoryIndexSearcher.java

```diff
@@ -214,6 +218,12 @@ public void transferCompleted(TransferEvent transferEvent) {
 		if (context == null) {
 			return CompletableFuture.runAsync(() -> { throw new IllegalArgumentException("context mustn't be null"); });
 		}
+		if ((context.getId().equals("https://repo.maven.apache.org/maven2") || context.getId().equals("central"))
+				&& disableCentralIndex) {
+			return CompletableFuture.runAsync(() -> {
+				System.out.println("Central repository index disabled");
```

Comment:
@mickaelistria do you have a suggestion for what the completable future should return instead of just printing something to stdout?